### PR TITLE
perf: change record dashboard

### PR DESCRIPTION
**Pull request by fit2bot**
perf: change record dashboard

### CODE REVIEW

File: apps/accounts/api/automations/change_secret.py [hunk 1]
```diff
@@ -75,11 +75,12 @@ def get_dashboard_queryset(queryset):
...
-        ).filter(Q(status=ChangeSecretRecordStatusChoice.success) | Q(ignore_fail=True))
+        ).filter(Q(status=ChangeSecretRecordStatusChoice.success))
 
         failed_records = queryset.filter(
             ~Q(account__in=Subquery(recent_success_accounts.values('account'))),
-            status=ChangeSecretRecordStatusChoice.failed
+            status=ChangeSecretRecordStatusChoice.failed,
+            ignore_fail=False
         )
         return failed_records
```
Comment (BaiJiangJie) [inline apps/accounts/api/automations/change_secret.py:86]: The difference between the original code (before the current date and time of writing this response) is that it filters only the records where the status is either successful or has ignored failures set to True. This change might impact certain functionality depending on the context of usage. A potential issue could be that there's no explicit mention in the code about when changes in the dashboard have been applied, which may complicate debugging efforts. For optimization suggestions:

1. Avoid unnecessary `if` statements: If not all conditions will always result in an execution, using an `else if` instead can optimize readability.

2. Use ORM for database queries: Using SQL query expressions makes the code less readable. Consider leveraging Django's ORM features like `filter` or `annotate`.

3. Check for dependencies: Ensure all required packages are imported before use. Inconsistent imports can lead to errors.

4. Code documentation: Detailed code comments describing what each section does helps maintain team coherence over time.

5. Error handling: Implement simple error messages rather than relying on generic "except" blocks. Debugging tools should still catch exceptions gracefully without overwhelming users with runtime warnings.
  
6. Testing strategy: Develop thorough unit tests across different scenarios and edge cases. Tests verify assumptions made in the development phase don't affect real-world application behaviors.

Remember these notes apply specifically based on given information and context at the moment I was written, thus suggesting further checks would typically include a review against up-to-date codebases.

File: apps/accounts/automations/base/manager.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def wait_and_save_recorder(self, recorder, max_retries=10, retry_interval=2):
 
             if exist:
                 print(f"Data inserted, updating recorder status after {attempt + 1}th query")
-                recorder.save(update_fields=['status', 'date_finished'])
+                recorder.save(update_fields=['error', 'status', 'date_finished'])
                 return True
 
             print(f"Data not ready, waiting {retry_interval} second(s) and retrying ({attempt + 1}/{max_retries})")
```
Comment (BaiJiangJie) [inline apps/accounts/automations/base/manager.py:150]: The given Python code does not contain any differences with current versions of Python, hence it is already updated and optimized to work properly.

This function waits for data that exists before saving the recorders' status. If the data doesn't appear within `max_retries` retries, another 2 seconds will be added until it's found again. The time units don't seem problematic based on their context here; they're likely specific to some external system or API you may be working with. As per current standards, such a setup should not encounter major errors unless the underlying system fails which would typically result in an error or a warning rather than returning False (which can mislead developers).

It seems like there aren't notable differences between this function from what we currently have available. Thus, no adjustments required here as everything appears correct for your provided information at the date listed.

Please let me know if you need further assistance or clarification!

**Review by feng626 — APPROVED** [review]
